protocol/result.go

```diff
@@ -18,13 +18,13 @@
 package protocol
 
 type Result interface {
-	SetError(error)
+	SetError(error) Result
```

Comment:
Why return Result? do you want  `SetError(e).SetResult(r)`? but you don't use this.？